sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/PayloadSizeGate.java

```diff
@@ -38,18 +38,12 @@
      * @return Buffered data or incoming data depending on threshold condition.
      */
     Flux<ByteBuffer> write(ByteBuffer buf) {
-        /* TODO (gapra): Investigate if there is way to avoid copying the data twice since this may result in lower
-            perf. The other copy is when we write the buffer to the pool */
-        /* Deep copy the buffer. This is required to ensure integrity of data. */
-        ByteBuffer cachedBuffer = ByteBuffer.allocate(buf.remaining()).put(buf);
-        cachedBuffer.flip();
```

Comment:
Is this change ok for cases where customer is interacting with async clients and using APIs that take Flux<ByteBuffer> instead of Streams?

Reply:
I think it should be. We only added this because of issues we were seeing in the sync case that came from an internal method reusing buffers to copy to an output stream. I don't think reusing buffers like that is a pattern in the async case because there aren't streams to copy between

Reply:
There are few tests that use `TransientFailureInjectingHttpPipelinePolicy` and use `Flux` (i.e. `Buffered upload handle pathing hot flux with transient failure`). However it's live only and it doesn't seem to cover larger payloads (i.e. it doesn't seem to cross single upload threshold). 
Could you add some coverage for this to be 100% sure?

Reply:
well, just noticed that it lowers single upload threshold. have you run it?

Reply:
I ran the live test pipeline and that seems to be passing. Only the analyze step looks to be failing atm